server/xpub-controller/helpers/files.js

```diff
@@ -33,22 +33,22 @@ class FilesHelper {
     }
   }
 
-  static startFileProgress(
+  static publishPredictedProgress(
     pubsub,
     ON_UPLOAD_PROGRESS,
     startedTime,
     predictedTime,
     manuscriptId,
   ) {
-    return setInterval(() => {
+    return () => {
```

Comment:
Do you think its worth putting some validation on the parameter here.... for example ...
```
preditctedTime > startedTime
manuscriptId.length == 36
typeof pubsub.publish == 'function' 
```
... just a question?

A bit of defensive programming here wouldn't go amiss seeing its the No1 experience for a user at the moment. Then we could check some failure cases too :)

Reply:
I like this idea. Then we could write some pretty simple tests ensuring an error is thrown when an invalid value is passed

Reply:
predicted time, is not a date, is just the time predicted in seconds, i.e 3.8907 seconds, so this condition is not valid, agree with the other ones.

Reply:
preditctedTime > 0 would work

Reply:
This would've been easier to add in a separate PR. It's making it more difficult to review at the moment given there's new functionality embedded in the refactor.